src/MIDebugEngine/Engine.Impl/ExceptionManager.cs

```diff
@@ -94,7 +96,7 @@ private class ExceptionCategorySettings
             public readonly ExceptionBreakpointStates DefaultCategoryState;
             public readonly ReadOnlyDictionary<string, ExceptionBreakpointStates> DefaultRules;
 
-            // Threading note: these are only read or updated by the FlushSettingsUpdates thread (in UpdateCatagory), and we
+            // Threading note: these are only read or updated by the FlushSettingsUpdates thread (in UpdateCategory) and TryGetExceptionBreakpoint, and we
```

Comment:
Update this comment to say: `// Since this is only written on the FlushSettingsUpdate thread, we can freely read this without a lock. However, all updates must take the lock.`

Reply:
I would suggest:

```
// Threading note: these are only read or updated by the FlushSettingsUpdates thread (in UpdateCategory) and TryGetExceptionBreakpoint. The following rules apply:
// 1. Writes and reads when *NOT* on the FlushSettingsUpdates thread - collection should be locked on itself
// 2. Reads on the FlushSettingsUpdates thread - no locking is needed
```